### PR TITLE
MLP Binary Classifier implemented

### PR DESCRIPTION
**Pull request by Saranga7**
Implemented an MLP based binary classifier with the already implemented (in river) loss functions, activations, and optimizers.

### CODE REVIEW

File: river/neural_net/mlp.py [hunk 1]
```diff
@@ -1,326 +1,416 @@
...
+
+    This model is still work in progress. Here are some features that still need implementing:
+
+    - `learn_one` and `predict_one` just cast the input `dict` to a single row dataframe and then
+        call `learn_many` and `predict_many` respectively. This is very inefficient.
+    - Not all of the optimizers in the `optim` module can be used as they are not all vectorised.
+    - Emerging and disappearing features are not supported. Each instance/batch has to have the
+        same features.
+    - The gradient haven't been numerically checked.
+
```
Comment (MaxHalford) [inline river/neural_net/mlp.py:344]: Could you add an `Examples` section, like `MLPRegressor`?

File: river/neural_net/mlp.py [hunk 2]
```diff
@@ -1,326 +1,416 @@
...
+        X
+            A DataFrame of shape (batch_size, n_features).
+
+        """
+        _, a = self._forward(X)
+        y_pred = a[self.n_layers]
+        return pd.DataFrame(y_pred, columns=self.targets, index=X.index)
+
+
+class MLPRegressor(base.Regressor, MLP):
```
Comment (MaxHalford) [inline river/neural_net/mlp.py:183]: Is there any reason GitHub displays this code as green (new)? What have you changed to `MLP` and `MLPRegressor`?

File: river/neural_net/mlp.py [hunk 3]
```diff
@@ -1,326 +1,416 @@
...
+
+    def learn_one(self, x, y):
+        # Multi-output
+        if isinstance(y, dict):
+            return self.learn_many(X = pd.DataFrame([x]), y=pd.DataFrame([y]))
+
+        # Single output
+        return self.learn_many(X = pd.DataFrame([x]), y=pd.Series([y]))
+
+    def predict_prob_one(self, x):
```
Comment (MaxHalford) [inline river/neural_net/mlp.py:402]:
```suggestion
    def predict_proba_one(self, x):

```

**Review by MaxHalford — CHANGES_REQUESTED** [review]
Thanks for contributing this. But it seems to me you haven't got much further than copy/pasting the code. I can only merge contributions that have done some testing. The fact that you wrote `predict_prob_one` instead of `predict_proba_one` tells me you haven't tested your implementation.

File: river/neural_net/mlp.py (hunk 1 above)
Comment (Saranga7) [reply river/neural_net/mlp.py:344]: Done

File: river/neural_net/mlp.py (hunk 2 above)
Comment (Saranga7) [reply river/neural_net/mlp.py:183]: No, I had copy pasted the whole mlp script from another local folder. I've only added the MLPBinaryClassifier code.

File: river/neural_net/mlp.py (hunk 3 above)
Comment (Saranga7) [reply river/neural_net/mlp.py:432]: Done

**Comment by MaxHalford** [comment]
I can't merge this till you fix the CI unit tests.

Also, what about multi-class classification? Users will expect an `MLPClassifier` to work with multi-class datasets.

**Comment by MaxHalford** [comment]
Closing because no news. Feel free to reopen and continue.

**Pull request closed without merging**